presto-main/src/main/java/io/prestosql/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -524,7 +540,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Expression> context)
                 rightSource = context.rewrite(node.getRight(), rightPredicate);
             }
 
-            Optional<Expression> newJoinFilter = Optional.of(combineConjuncts(metadata, joinFilterBuilder.build()));
+            Optional<Expression> newJoinFilter = Optional.of(combineConjuncts(metadata, dynamicFiltersResult.getJoinFilterClauses()));
```

Comment:
`getJoinFilterClauses` is not needed. We should always add `joinFilterBuilder.build()` expressions (regardless of DF).

Reply:
Good catch, thanks!